cpp/src/arrow/ipc/message.h

```diff
@@ -441,6 +441,10 @@ class ARROW_EXPORT MessageReader {
   virtual Result<std::unique_ptr<Message>> ReadNextMessage() = 0;
 };
 
+// the first parameter of the function should be a pointer to metadata (aka.
+// org::apache::arrow::flatbuf::RecordBatch*)
+using FieldsLoaderFunction = std::function<Status(const void*, io::RandomAccessFile*)>;
```

Comment:
from CI, it seems we need `#include <functional>` in this header now.

Reply:
Hmmm, somehow this error is not reported when I compiled/ran the tests locally. I got it fixed in the latest commit.